### PR TITLE
Add optical track initialization action

### PR DESCRIPTION
**Pull request by amandalund**
This add the track initialization action for the optical loop as well as optical simulation and particle data and views. I'll follow up with the action to locate vacancies.

### CODE REVIEW

File: src/celeritas/optical/CoreTrackView.hh [hunk 1]
```diff
@@ -88,6 +96,42 @@ CoreTrackView::CoreTrackView(ParamsRef const& params,
...
+    }
+
+    // Initialize the particle state
+    this->particle()
+        = ParticleTrackView::Initializer{init.energy, init.polarization};
+
+    //! \todo Add physics view and clear physics state
+
+    return *this;
+}
```
Comment (sethrj) [inline src/celeritas/optical/CoreTrackView.hh:133]: I like it! Future work for the main core track view?  (And would we need a second version for the "detailed initializer" that has the full volume state?)

File: src/celeritas/optical/interactor/RayleighInteractor.hh [hunk 2]
```diff
@@ -37,8 +37,8 @@ class RayleighInteractor
 {
   public:
     // Construct interactor from an optical track
-    inline CELER_FUNCTION
-    RayleighInteractor(TrackView const& particle, Real3 const& direction);
+    inline CELER_FUNCTION RayleighInteractor(ParticleTrackView const& particle,
+                                             Real3 const& direction);
```
Comment (sethrj) [inline src/celeritas/optical/interactor/RayleighInteractor.hh:41]: Clang-format differences 😒 really gotta get this into the CI, don't worry about it here

**Review by sethrj — APPROVED** [review]
Excellent!

File: src/celeritas/optical/CoreTrackView.hh (hunk 1 above)
Comment (amandalund) [reply src/celeritas/optical/CoreTrackView.hh:133]: Yep! And yeah, gotta think about the best way to handle the special geo initialization...

**Comment by amandalund** [comment]
Actually @sethrj if you want to hold off merging this, I might just add the "locate vacancies" to this track initialization action and refactor that part once we're producing secondaries.

**Comment by sethrj** [comment]
Sure thing! Separate PR works just as well :)

**Comment by amandalund** [comment]
Ok cool, then maybe I'll just do it separate since I think we do want it at the end of the loop so the counters are updated correctly.

**Pull request merged**